sopel/modules/reddit.py

```diff
@@ -25,27 +27,13 @@
 domain = r'https?://(?:www\.|old\.|pay\.|ssl\.|[a-z]{2}\.)?reddit\.com'
 post_url = r'%s/r/(.*?)/comments/([\w-]+)' % domain
 user_url = r'%s/u(ser)?/([\w-]+)' % domain
-post_regex = re.compile(post_url)
-user_regex = re.compile(user_url)
 spoiler_subs = [
     'stevenuniverse',
     'onepunchman',
 ]
 
 
-def setup(bot):
-    if not bot.memory.contains('url_callbacks'):
-        bot.memory['url_callbacks'] = SopelMemory()
-    bot.memory['url_callbacks'][post_regex] = rpost_info
-    bot.memory['url_callbacks'][user_regex] = redditor_info
-
-
-def shutdown(bot):
-    del bot.memory['url_callbacks'][post_regex]
-    del bot.memory['url_callbacks'][user_regex]
-
-
-@rule('.*%s.*' % post_url)
+@url(post_url)
 def rpost_info(bot, trigger, match=None):
```

Comment:
The `match` parameter should be always provided when `@url` is used instead of `@rule`.

```suggestion
def rpost_info(bot, trigger, match):
```